### DISCUSSION TITLE
Cannot provide a value for property 'IdGenerator'

### DISCUSSION THREAD

**Discussion by mindaugas-epoint** [Megabit/Blazorise#2172] [Q&A] [answered]
Trying to run LineChart example provided [here](https://blazorise.com/docs/extensions/chart/) in my Blazor app and getting this error:

InvalidOperationException: Cannot provide a value for property 'IdGenerator' on type 'Blazorise.Charts.LineChart`1[[System.Double, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]'. There is no registered service of type 'Blazorise.IIdGenerator'.

Any idea?

**Comment by stsrki** [comment, marked as answer]
Maybe you missed the part with registering Blazorise components. `.AddBlazorise()` ?

https://blazorise.com/docs/start/#4a-blazor-webassembly

**Reply by mindaugas-epoint** [reply]
Yes, I've missed this part. Thank You.

**Reply by Xeevis** [reply]
I'll just add that in case someone uses prerendering, it's necessary to add `.AddBlazorise()` in both WebAssembly and host projects. Caught me a little off-guard 😅.

**Reply by stsrki** [reply]
Haha that was always a requirement 😅
